src/paths/steiner.cpp

```diff
@@ -0,0 +1,535 @@
+			
+			
+			igraph_vector_int_append(vectorlist_all,&vectorlist_1);
+			igraph_vector_int_append(edgelist_all,&edgelist_1);
+
+		
+			igraph_vector_int_destroy(&vectorlist_1);
+			igraph_vector_int_destroy(&edgelist_1);
+			IGRAPH_FINALLY_CLEAN(2);
+
+		}
+		else {
+			
+			igraph_integer_t E1,F1;
+
+			E1 = *D.begin();
+			F1 = *next(D.begin(),1);;	
+			
+			//std::cout << "E1:" <<E1 << std::endl;
+
+			igraph_vector_int_t vectorlist_1;
+			IGRAPH_CHECK(igraph_vector_int_init(&vectorlist_1,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&vectorlist_1);
+
+			igraph_vector_int_t edgelist_1;
+			IGRAPH_CHECK(igraph_vector_int_init(&edgelist_1,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&edgelist_1);
+
+			igraph_get_shortest_path_dijkstra(graph,&vectorlist_1,&edgelist_1,k,E1,weights,IGRAPH_ALL);
+
+			igraph_vector_int_t vectorlist_2;
+			IGRAPH_CHECK(igraph_vector_int_init(&vectorlist_2,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&vectorlist_2);
+
+			igraph_vector_int_t edgelist_2;
+			IGRAPH_CHECK(igraph_vector_int_init(&edgelist_2,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&edgelist_2);
+
+			igraph_get_shortest_path_dijkstra(graph,&vectorlist_2,&edgelist_2,k,F1,weights,IGRAPH_ALL);
+
+			igraph_vector_int_append(vectorlist_all,&vectorlist_1);
+			igraph_vector_int_append(vectorlist_all,&vectorlist_2);
+
+			igraph_vector_int_append(edgelist_all,&edgelist_1);
+			igraph_vector_int_append(edgelist_all,&edgelist_2);
+
+			igraph_vector_int_destroy(&vectorlist_2);
+			igraph_vector_int_destroy(&vectorlist_1);
+			
+			igraph_vector_int_destroy(&edgelist_1);
+			igraph_vector_int_destroy(&edgelist_2);
+
+			IGRAPH_FINALLY_CLEAN(4);
+
+			std::set<igraph_integer_t> min_F;
+			min_F.insert(F1); 
+
+			//std::cout << "Got till here!" << std::endl;
+
+		}
+		
+		m = k;
+		D = min_F;			
+		
+		igraph_vector_int_destroy(&vectorlist);
+		igraph_vector_int_destroy(&edgelist);
+		IGRAPH_FINALLY_CLEAN(2);
+	}
+	
+
+
+	return IGRAPH_SUCCESS;
+
+}
+
+
+igraph_error_t igraph_steiner_dreyfus_wagner(const igraph_t *graph,const igraph_vector_int_t* steiner_terminals,
+igraph_neimode_t mode, const igraph_vector_t *weights,igraph_real_t *res,igraph_vector_int_t *res_tree)
+{
+	if (mode != IGRAPH_ALL)
+	{
+		std::cout << "Currently this function only supports undirected graphs while the graph's mode is not undirected." <<std::endl;
+		return IGRAPH_FAILURE;
+	}
+	
+	
+	igraph_integer_t no_of_vertices = (igraph_integer_t)igraph_vcount(graph);
+	igraph_integer_t no_of_edges = igraph_ecount(graph);
+
+	// if (igraph_vector_int_size(steiner_terminals) == no_of_vertices)
+	// {
+	// 	std::cout << "Getting Minimum Spanning Tree" << std::endl;
+	// 	igraph_error_t ans = igraph_minimum_spanning_tree(graph,res,weights);
+	// 	return ans;
+		
+	// } Needs to be moved to Phase 2 when we implement backtracking to get the edges out
+	if (no_of_vertices == 0 || (no_of_vertices == 1)) //graph is empty
+	{
+		*res = 0;
+		return IGRAPH_FAILURE;
```

Comment:
Do not return a non-success error code without also calling the error handler. In other words, use the `IGRAPH_ERROR` or `IGRAPH_ERRORF` macros.